[ADD] add 'OCA/repository' as category when sorting addons coming from OCA

AnizR opened the pull request:
### Issue

- [x] fixes issue #14 

### Description

Instead of having every OCA addon under the category '#OCA', it sorts by github repository (e.g. #OCA/account-payment)

Commit:
[ADD] add 'OCA/repository' as category when sorting addons coming from OCA

AnizR commented:
ping @marielejeune @qgroulard

sbidoul commented on README.md:
```diff
@@ -21,7 +21,10 @@ Options:
+  --oca-category                Add category for third party addons coming from OCA.
+                                The category is set as 'OCA/repository_name'.
+                                If the repository can not be identified,
+                                it falls into the default 'OCA' category.
```
It would be nice if we could have the grouping by OCA repo optional, as I personally prefer to have a single category for OCA.

AnizR replied:
Done, we have 'three levels of categories':
1) not setting '--oca-cateogry': oca addons fall into 'Third-party' category
2) '--oca-cateogry=basic': oca addons are grouped under 'OCA' category
3) '--oca-category=repository': oca addons are grouped by repository 'OCA/<repository_name>'

sbidoul commented on src/odoo_sort_manifest_depends/sort_manifest_deps.py:
```diff
@@ -53,17 +65,65 @@ def _identify_oca_addons(addon_names: list[str], odoo_series: OdooSeries) -> tup
+
+    if not repo_url:
+        return None
+    return search("OCA/.*", repo_url).group()
```
```suggestion
    return re.search("OCA/.*", repo_url).group()
```

When the function name is generic like that, I find it easier to have the module name here. I actually had to scroll up to understand what `search` was here.

Also take care that re.search can return None.

AnizR committed:
[CHG] do not cache addons that weren't found and classified as 'other'

yvaucher committed:
[ADD] tests on cache management

Add optional `cache` parameter to `_identify_oca_addons()` in order
to inject a test cache and not mess with local cache.

yvaucher committed:
[IMP] do not cache OCA modules without categories

If a module exists in the wheelhouse index but not in the current
version it can be categorized as OCA, however the location is unsure.

Thus such OCA module shouldn't be added in the cache as we want this
tool to find out when it finally enters in the wheelhouse.

Usually it means you have added a dependency to your project that
has not been merged on github.com/OCA

AnizR commented:
Thanks for the commits @yvaucher :rocket:

AnizR reviewed (commented):
I have been using this PR for a while now, I think that I'll merge it
